tests/thread_float/Makefile

```diff
@@ -3,6 +3,7 @@ include ../Makefile.tests_common
 USEMODULE += printf_float
 USEMODULE += xtimer
 
-#DISABLE_MODULE += cortexm_fpu
+# Use CortexM FPU when available (only on M4F and M7 CPUs)
+FEATURES_OPTIONAL += cortexm_fpu
```

Comment:
So the FPU only gets used in this test (or when explicitly requested), otherwise software emulation is used.
What is the benefit of that?

If you have an FPU that should always be preffered over `float-abi=soft`, no?

Reply:
Good point, I'll update the PR in that direction !

Reply:
I've been trying and I realized that to be possible it would need #9913 to be fixed first.
This is because cortexm.inc.mk is included before processing the dependency resolution and this PR relies on this mechanism. Even if we would keep the FPU by default (using a default module), that would still be required (this is what I was thinking of and tried actually).

> If you have an FPU that should always be preffered over float-abi=soft, no?

Yes probably.

I think I'll rework that PR and cut the CPU_ARCH/FAM move out of it.